### ISSUE TITLE
MD051: False-positive if the header begins with `--`

### ISSUE THREAD

**Issue by mrazauskas** [DavidAnson/markdownlint#960]
The following:

```md
## `--config`

The path to a configuration file.

See [`--config`](#config).
```

Gets reported as: MD051/link-fragments: Link fragments should be valid

Did I miss something? (;

Playground link: https://dlaa.me/markdownlint/#%25m%23%20Docs%0A%0A%23%23%20%60--config%60%0A%0AThe%20path%20to%20a%20configuration%20file.%0A%0ASee%20%5B%60--config%60%5D(%23config)%0A

**Comment by DavidAnson** [comment]
GitHub rules translate that heading into: #--config

If you update your demonstration to use that, no warning is produced.

**Comment by mrazauskas** [comment]
Thanks. Yes, your suggestion works as expected.

I was misguided by VS Code autocompletions:

<img width="383" alt="Screenshot 2023-09-09 at 07 19 52" src="https://github.com/DavidAnson/markdownlint/assets/72159681/5e5d4baf-8980-4639-b5e7-1d1802e53e7e">
